### ISSUE TITLE
Won't finish loading on macOS Sierra with mini player option disabled

### ISSUE THREAD

**Issue by oojacoboo** [radiant-player/radiant-player-mac#600]
The app gets to the following image before halting.  I'm using macOS Sierra.  This seems to be happening with "Show mini player in menu bar" disabled, which prevents the app from crashing when enabled.

http://files.jacobt.com/ZHmMdgeo6E

Also, during the loading up process, I can see multiple windows/UI components flash quickly at the top left of the application window.  I'm assuming something is loading in the background, but it's quite strange.

**Comment by tjanson** [comment]
Duplicate of #589. Look in the comments there for a working build. (I had the same issue today, I think Google rolled out those changes to everyone today.)

**Comment by oojacoboo** [comment]
It's actually broken in many ways.  I tried some of the builds from #589 and those don't really get me anywhere.  The mini-player doesn't work at all and is related to #574.

We need some real testing and fixing on macOS Sierra.  There are going to be a lot of people on Sierra soon.

**Comment by p-becker** [comment]
Ran into the same crash today. Build 007 from #589 worked for me:
https://dl.dropboxusercontent.com/u/6278910/RadiantTest/007.zip

However, I am still on OS X El Capitan and in my case, changing the mini player settings did not have any effect.

**Comment by BarakaAka1Only** [comment]
I just started to get that myself today, pretty much the drawer navigation in the navigation.js in Radiant Player isn't allowing the selector on load and pushes a non-stop error even though the item is there

![image](https://cloud.githubusercontent.com/assets/2947494/18516458/b06dd6cc-7a5e-11e6-88f0-d605aeb2d16d.png)

if you simply comment out 

```
 // Adjust the drawer navigation scrolling mode to accomodate the traffic light buttons.
    var el = document.querySelector('#drawer #nav-container');
    if (el) el.setAttribute('mode', '');

```

to

```
 // Adjust the drawer navigation scrolling mode to accomodate the traffic light buttons.
    //var el = document.querySelector('#drawer #nav-container');
    //if (el) el.setAttribute('mode', '');
```

then it works tested on Mavericks, Yosemite, and El Capitan still not sure why that is but as to https://github.com/radiant-player/radiant-player-mac/issues/589 it may not just be the navagation just like for myself.

**Comment by tjkirch** [comment]
Thanks BarakaAka1Only!  Your code change fixes Radiant launch for me, and I haven't noticed any problems.

Note: I'm using Radiant 1.10.2 on Mac OS 10.10.5 -- so it's not Sierra specific.

**Comment by BarakaAka1Only** [comment]
Great to hear it @tjkirch :)

**Comment by jacobwgillespie** [comment]
Thanks @BarakaAka1Only, I've been meaning to look into this but have been AFK moving this past week.  I'll deploy a hotfix with your changes (#604) and we can see how many issues that cleans up.

👍

**Comment by BarakaAka1Only** [comment]
@jacobwgillespie no problem and no worries you got here right on time ;)

**Comment by jacobwgillespie** [comment]
Can you check on v1.10.3?

**Comment by tjkirch** [comment]
Working great for me on v1.10.3, thanks!

**Comment by oojacoboo** [comment]
Technically this fix does load up the player now.  But there are many other things still bjorked on macOS Sierra.  It's certainly not usable.  #574 still exists, and I also now get the following...

![screen shot 2016-09-14 at 9 46 28 pm](https://cloud.githubusercontent.com/assets/764664/18535711/b7dcae7c-7ac4-11e6-9017-e35a07077dba.png)

**Comment by BarakaAka1Only** [comment]
@oojacoboo and to all with macOS Sierra, gonna get myself a copy of macOS Sierra and will take a look to see what is going on even with https://github.com/radiant-player/radiant-player-mac/issues/574 will update here once i have something

**Comment by BarakaAka1Only** [comment]
Okay finally gotten a chance to take a look and as to @oojacoboo's and https://github.com/radiant-player/radiant-player-mac/issues/574 issue, The CookieStorage isn't settings all the cookies.

When i first checked this all the cookies when in the web inspector  were blank and then suddenly 
it only sets the doubleclick.net and one cookie for logging in but not the SID's and everything else that Google Play needs to function so what i did was added a check.

if the OS is macOS Sierra to just allow the request to HandleCookies, but this is the same as going into the preferences -> Privacy Tab -> and checking on Use Safari's Cookies but some users can miss that or don't bother playing with the  preferences which is why i added a "force" pretty much


This check also applies for the MiniPlayer as BAD_ACCESS occurs in the updateTrackingAreas method when trying to remove the tracking area

so the following

`
if(!delegate.appDelegate.isSierra) {
 //allow only El Capitan and below to remove trackingArea for now
}
`

_isSierra is an added BOOL in appDelegate_

but this is __**not a fix**__, sometimes clearColor may happen or when the MiniPlayer Panel is shown and then hidden their maybe a white window, I did this on a VM so i am not sure if that's the reason why
but if you get that just relaunch (if clearColor happens you can play a song and the album art will at least add a background and work like normal-ish)

I will still look into it more when i get a chance but this is just to allow the Sierra users to use Radiant Player for now.

Let me know if it works
_you may need to re-sign in most likely not_ 

Download: 

https://www.dropbox.com/s/u9ksdk3j83ow9ch/macOS%20Sierra%20Radiant%20Player.zip?dl=0

**Comment by smackjack3** [comment]
Thank you @BarakaAka1Only it's working now for me.

**Comment by oojacoboo** [comment]
Good work @BarakaAka1Only - seems to be mostly all working now.  I'll continue to look for any bugs.  But this certainly helps get things operational for now!

**Comment by BarakaAka1Only** [comment]
No problem ;)

**Comment by jacobwgillespie** [comment]
Thanks @BarakaAka1Only, send over a PR and I'll get it merged :)

**Comment by BarakaAka1Only** [comment]
@jacobwgillespie  Will do, right now i am doing some style updates for the css once that is done i'll do a PR :)

**Comment by BarakaAka1Only** [comment]
Okay @jacobwgillespie looked into the MiniPlayer issue more on Sierra and when i was about to do a PR wanted to ensure everything works. The first time i did use Update recommended settings, and that allowed the semi fix to work, but i didn't do it this time cause its not recommended to do that if i am gonna PR (should haven't the first time)

So not doing so, brings the same issue but another issue so with the original `options:`.
It wasn't getting pushed/set via int from `int opts` so i did a condition if on El Capitan and below to use the original  code while
Sierra doesn't remove the trackingArea instead adds it with the options not as an int. 
and with that everything works like it should :D 

### **Visual**


```
-(void)updateTrackingAreas {

    if (trackingArea != nil) {
        if(!delegate.appDelegate.isSierra) {
            [self removeTrackingArea:trackingArea];
        }
    }

    if(!delegate.appDelegate.isSierra) {
        int opts = (NSTrackingMouseEnteredAndExited | NSTrackingActiveAlways);
        trackingArea = [ [NSTrackingArea alloc] initWithRect:[self bounds]
                                                 options:opts
                                                   owner:self
                                                userInfo:nil];
        [self addTrackingArea:trackingArea];
    } else {
        NSTrackingArea* trackingArea = [[NSTrackingArea alloc]
                                            initWithRect:[self bounds]
                                            options: (NSTrackingMouseEnteredAndExited |  NSTrackingActiveAlways)
                                            owner:self
                                            userInfo:nil];
        [self addTrackingArea:trackingArea];
    }

}
```

This also includes CSS updates for all the custom styles :) 

### Download:
https://www.dropbox.com/s/r989wiympa1b4jv/macOS%20Sierra%20Radiant%20Player%20v2.zip?dl=0

_This will fix https://github.com/radiant-player/radiant-player-mac/issues/574 with no glitches from what i said from the first version with the semi fix https://github.com/radiant-player/radiant-player-mac/issues/600#issuecomment-247295337_ 

Please anyone with macOS Sierra please confirm (@smackjack3, @oojacoboo, @Shmuwol) so that i can push a PR ;)

w00t for happy coding 👍 

-Baraka

**Comment by oojacoboo** [comment]
@BarakaAka1Only tested and is working well from what I can tell thus far.  

Not sure if this is new, or an existing issue, but upon first load, within the mini player, you can't hit play - no playlist or anything selected.  You have to load up the main player and choose a playlist, then the play options work.  This isn't a huge issue, but it's a thing.

Also, along these same lines, there isn't any hover or selected event styling on the mini player buttons.  Again, not sure if this is new, or has always been the case, but issues IMO, nonetheless.

**Comment by BarakaAka1Only** [comment]
@oojacoboo

From what i know and myself with playing around with the code, on first load you can't just hit play and expect something to play, this is because on load MUSIC_PLAYING is false which is correct once you play an item on the main player OR Roll the dice in the Mini Player you will be able to play/pause/skip everything is connected ;)

The buttons don't have a hover or a :active/focus if you click on a button and don't let go you will see it grey out, basically the NSButtons there are images and with the events for mouseEntered /Exited
i don't think its possible for every individual button to have its own event for a hover and will be more in memory i suppose. 

I've just gotten into coding on objective-c and i'm learning a lot everyday and really enjoying it from that i've been working on a re-amp on the Mini Player with added stuff. Right now its not ready but soon i'll have something :)

Hope this helps!

**Comment by oojacoboo** [comment]
@BarakaAka1Only  Rock on!

**Comment by johnpane** [comment]
@BarakaAka1Only your v2 fixes "Error loading library" on Sierra for me

**Comment by mordocai** [comment]
@BarakaAka1Only Just chiming in since I don't see the promised pull request yet. Your v2 download fixes #608 for me as well.

**Comment by BarakaAka1Only** [comment]
Great to hear @mordocai @johnpane and thank you for testing, sorry about the waited time on this, a PR will be submitted today!
